.github/workflows/build-preview.yml

```diff
@@ -39,10 +39,10 @@ jobs:
       - name: Capture main branch sample clips
         id: main-samples
         run: |
-          npm start & command_pid=$!
-          sleep 15
+          npm start &
+          while ! nc -z localhost 1776; do sleep 1; done
           npm run generate_samples
-          kill $command_pid
+          kill $(lsof -t -i:1776) 2>/dev/null || echo "No process found listening on port 1776."
```

Comment:
man bash
/jobspec

`kill %`

Reply:
I found that the PID of `npm start` is not actually the PID of the :1776 web server, which is why I changed it to this lsof business.